src/Core/Parser/BooleanParser.php

```diff
@@ -0,0 +1,545 @@
+<?php
+namespace TYPO3Fluid\Fluid\Core\Parser;
+
+/*
+ * This file belongs to the package "TYPO3 Fluid".
+ * See LICENSE.txt that was shipped with this package.
+ */
+
+/**
+ * This BooleanParser helps to parse and evaluate boolean expressions.
+ * it's basically a recursive decent parser that uses a tokenizing regex
+ * to walk a given expression while evaluating each step along the way.
+ *
+ * For a basic recursive decent exampel check out:
+ * http://stackoverflow.com/questions/2093138/what-is-the-algorithm-for-parsing-expressions-in-infix-notation
+ *
+ * Parsingtree:
+ *
+ * 	parse
```

Comment:
This looks a little off - tabs/spaces issue? Also, I'm not sure how illustrative this tree is. It might help if it starts with an actual expression and notes each matched part in this tree in the branch where it gets parsed/evaluated.